### PR TITLE
Desktop: Add toggling functionality for bold, italics and code in Editor Toolbar

### PR DESCRIPTION
**Pull request by rabeehrz**
Fixes issue #2560 

This PR adds a toggling function for bold, italics and code elements in the Editor Toolbar.

As mentioned in the issue:

> You have a text `bla`, select it and click `Bold`. Then you get `**bla**`, you select `**bla**`, click `Bold` and get `bla`.

#### Improvements

Added italics toggling for wrapping using underscores

Italics

### Before

Before

### After

After

### CODE REVIEW

**Comment by rabeehrz** [comment]
@PackElend label me please.

File: ElectronClient/gui/NoteText.jsx [hunk 1]
```diff
@@ -1512,11 +1512,23 @@ class NoteTextComponent extends React.Component {
...
+		} else {
+			this.wrapSelectionWithStrings('**', '**', _('strong text'));
+		}
 	}
 
 	commandTextItalic() {
-		this.wrapSelectionWithStrings('*', '*', _('emphasized text'));
+		const selection = this.textOffsetSelection();
+		let string = this.state.note.body.substr(selection.start, selection.end - selection.start);
+		if (string.startsWith('*') && string.endsWith('*')) {
```
Comment (tessus) [inline ElectronClient/gui/NoteText.jsx]: Please also check for underscore here.
`if ((string.startsWith('*') && string.endsWith('*')) || (string.startsWith('_') && string.endsWith('_'))) {`

Comment (rabeehrz) [reply ElectronClient/gui/NoteText.jsx]: @tessus I've made the requested changes and update the issue. Please review.

**Comment by tessus** [comment]
There's one line between the code block and the text.

cb

Now there are 2 empty lines bwtween the code block and the text.

**Comment by rabeehrz** [comment]
> use this in commandTextCode():
> 
> ```
>                      if (string.startsWith('```') && string.endsWith('```')) {
> 			this.wrapSelectionWithStrings('', '', '', string.substr(4, selection.end - selection.start - 8
> 		} else {
> 			this.wrapSelectionWithStrings(`\`\`\`${match[0]}`, `${match[0]}\`\`\``);
> 		}
> ```
> 
> This will fix the extra line appearing.

Thanks, @meyash 
@tessus I've added the above change and that seemed to have solved the issue. Please review.

**Comment by laurent22** [comment]
The four blocks of code you've added are all very similar. Could you refactor them into a single function please?

**Comment by rabeehrz** [comment]
> The four blocks of code you've added are all very similar. Could you refactor them into a single function please?

I've shortened the code a bit without making it very complicated.
I've used the old style for code block replacement because the two cases (single line and multiple line) should be handled separately.

**Comment by tessus** [comment]
Anything else that has to be done or can we merge?

**Comment by laurent22** [comment]
Looks good, thanks @rabeehrz!

**Comment by laurent22** [comment]
Did this feature ever worked? On the latest version if I select some text and click on bold it becomes `**some text**`, then if I click again it becomes `****some text****`, so like it always did. Not sure if the feature got broken or if it had some bug. For now, I'm refactoring the text editor component and removing this feature as I don't know how to fix it.

**Comment by tessus** [comment]
Yes, You have to select the text first.

But there's also a follow-up PR which I think does what you think it should do.

**Comment by tessus** [comment]
#3077

**Pull request merged**